nix/project.nix

```diff
@@ -0,0 +1,70 @@
+{ inputs, pkgs, lib }:
+
+let
+  cabalProject = pkgs.haskell-nix.cabalProject' (
+
+    { config, pkgs, ... }:
+
+    {
+      name = "kes-agent";
+
+      compiler-nix-name = lib.mkDefault "ghc967";
+
+      src = lib.cleanSource ../.;
+
+      flake.variants = {
+        ghc967 = { };
+
+        # Disabled other compilers for now, uncomment if needed
+        # but the dependencies pinned in the kes-agent.cabal 
+        # file are to strict for these versions
+
+        # ghc984.compiler-nix-name = "ghc984";
+        # ghc9102.compiler-nix-name = "ghc9102";
+        # ghc9122.compiler-nix-name = "ghc9122";
+      };
+
+      inputMap = { "https://chap.intersectmbo.org/" = inputs.CHaP; };
+
+      modules = [
+        ({ pkgs, lib, config, ... }:
+          let
+            # A small helper function to generate post-install
+            # scripts for CLI tools to add completions
+            # to bash and zsh.
+            postInstallCompletion = exeName:
+              "  BASH_COMPLETIONS=$out/share/bash-completion/completions\n  ZSH_COMPLETIONS=$out/share/zsh/site-functions\n  mkdir -p $BASH_COMPLETIONS $ZSH_COMPLETIONS\n  $out/bin/${exeName} --bash-completion-script ${exeName} > $BASH_COMPLETIONS/${exeName}\n  $out/bin/${exeName} --zsh-completion-script ${exeName} > $ZSH_COMPLETIONS/_${exeName}\n";
+          in
+          # we need git at compile time since we use git to generate
+            # the version number in the binary
+            # Try `nix run .#kes-agent -- --version'
+          lib.mkIf (!pkgs.stdenv.hostPlatform.isWindows) {
```

Comment:
@geo2a , wdyt: given [this](https://github.com/perturbing/kes-agent/commit/6276eeae24e0a34947cf38f67277fca563cf2afa) implicit change, shall we remove this if statement?

Reply:
I think this is beyond the boundary of my Nix knowledge :smile: I've simply replicated the [setup](https://github.com/IntersectMBO/ouroboros-consensus/blob/34a5f5b5bfc2ee0231b83e0cab78f78a8e2b6b41/flake.nix#L58) we have in Consensus.

My reasoning was: we'd like to build and test on the same systems that we do the Consensus layer, and it does not make much sense to do more. `cardano-node` also only lists these three systems, see https://github.com/IntersectMBO/cardano-node/blob/master/nix/supported-systems.nix. I assume the testing on Windows is done by some non-Nix means.

So I think it's OK to eliminate Windows-specific stuff from the flake until we're asked to add it.